Can't build on stable (?)

Seems surprising, but:

```
   Compiling email_address v0.2.9
error[E0554]: `#![feature]` may not be used on the stable release channel
   --> /home/mgh/.cargo/registry/src/index.crates.io-6f17d22bba15001f/refined-0.3.0/src/lib.rs:377:1
    |
377 | #![feature(doc_cfg)]
    | ^^^^^^^^^^^^^^^^^^^^

   Compiling picky-asn1 v0.10.1 (/home/mgh/code/projects/picky-rs-sp1ff/picky-asn1)
...
For more information about this error, try `rustc --explain E0554`.
error: could not compile `refined` (lib) due to 1 previous error
```

I confess I'm new to the `feature` attribute, but according to the [Cargo Book](https://doc.rust-lang.org/reference/attributes.html) it is "used to enable unstable or experimental compiler features". Is this intended?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,11 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
 
+## [0.3.1] - 2025-03-30
+
+- Fixes an inability to build using the `stable` toolchain caused by indiscriminate usage of the
+  unstable `doc_cfg` feature
+
 ## [0.3.0] - 2025-03-25
 
 - Adds the `alloc` feature flag, providing support for `no_std` without requiring the use of an

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "refined"
-version = "0.3.0"
+version = "0.3.1"
 edition = "2021"
 description = "Simple refinement types; parse, don't validate!"
 documentation = "https://docs.rs/refined"
@@ -30,3 +30,4 @@ optimized = []
 
 [package.metadata.docs.rs]
 all-features = true
+rustdoc-args = [ "--cfg", "docsrs" ]

--- a/flake.nix
+++ b/flake.nix
@@ -28,6 +28,7 @@
         pkgs = nixpkgs.legacyPackages.${system};
         fenix' = fenix.packages.${system};
         crane' = (crane.mkLib pkgs).overrideToolchain fenix'.complete.toolchain;
+        crane-stable = (crane.mkLib pkgs).overrideToolchain fenix'.stable.minimalToolchain;
         manifest = (pkgs.lib.importTOML ./Cargo.toml).package;
 
         src = crane'.cleanCargoSource ./.;
@@ -60,6 +61,11 @@
         refined = crane'.buildPackage {
           inherit src;
           cargoTestExtraArgs = "--all-features";
+        };
+
+        refined-stable = crane-stable.buildPackage {
+          inherit src;
+          cargoTestExtraArgs = "--lib";
         };
 
         refined-no-std = crane'.buildPackage {
@@ -109,11 +115,14 @@
             linuxPackages_latest.perf
             lldb
           ];
+
+          RUSTDOCFLAGS = "--cfg docsrs";
         };
 
         checks.${system} = {
           inherit
             refined
+            refined-stable
             refined-no-std
             refined-doc
             refined-example-axum

--- a/src/implication/mod.rs
+++ b/src/implication/mod.rs
@@ -50,10 +50,10 @@ impl IsTrue for Assert<true> {}
 mod boolean_imp;
 mod boundable_imp;
 
-#[doc(cfg(feature = "arithmetic"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "arithmetic")))]
 #[cfg(feature = "arithmetic")]
 mod arithmetic;
 
-#[doc(cfg(feature = "arithmetic"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "arithmetic")))]
 #[cfg(feature = "arithmetic")]
 pub use arithmetic::*;

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -374,7 +374,7 @@
     allow(incomplete_features),
     feature(generic_const_exprs)
 )]
-#![feature(doc_cfg)]
+#![cfg_attr(docsrs, feature(doc_cfg))]
 #![cfg_attr(not(feature = "std"), no_std)]
 
 #[cfg(feature = "alloc")]
@@ -393,7 +393,7 @@ pub mod boolean;
 pub mod boundable;
 pub mod character;
 pub mod prelude;
-#[doc(cfg(feature = "alloc"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
 #[cfg(feature = "alloc")]
 pub mod string;
 
@@ -403,10 +403,10 @@ pub use refinement::*;
 pub use boundable::signed::SignedBoundable;
 pub use boundable::unsigned::UnsignedBoundable;
 
-#[doc(cfg(feature = "implication"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "implication")))]
 #[cfg(feature = "implication")]
 pub mod implication;
-#[doc(cfg(feature = "implication"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "implication")))]
 #[cfg(feature = "implication")]
 pub use implication::*;
 
@@ -548,7 +548,7 @@ pub struct Refined<T>(T);
 pub struct RefinementError(ErrorMessage);
 
 #[cfg(feature = "alloc")]
-#[doc(cfg(feature = "alloc"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
 impl Display for RefinementError {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
         write!(f, "refinement violated: {}", self.0)

--- a/src/refinement/mod.rs
+++ b/src/refinement/mod.rs
@@ -3,7 +3,7 @@ mod named;
 
 use core::{fmt::Display, marker::PhantomData};
 
-#[doc(cfg(feature = "alloc"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
 #[cfg(feature = "alloc")]
 pub use named::*;
 
@@ -23,7 +23,7 @@ use crate::Implies;
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
 pub struct Refinement<T, P: Predicate<T>>(pub(crate) T, pub(crate) PhantomData<P>);
 
-#[doc(cfg(feature = "serde"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
 #[cfg(feature = "serde")]
 impl<T: Serialize, P: Predicate<T>> Serialize for Refinement<T, P> {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
@@ -34,7 +34,7 @@ impl<T: Serialize, P: Predicate<T>> Serialize for Refinement<T, P> {
     }
 }
 
-#[doc(cfg(all(feature = "serde", feature = "alloc")))]
+#[cfg_attr(docsrs, doc(cfg(all(feature = "serde", feature = "alloc"))))]
 #[cfg(all(feature = "serde", feature = "alloc"))]
 impl<'de, T: Deserialize<'de>, P: Predicate<T>> Deserialize<'de> for Refinement<T, P> {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
@@ -102,7 +102,7 @@ impl<T, P: Predicate<T>> TryFrom<Refined<T>> for Refinement<T, P> {
     }
 }
 
-#[doc(cfg(feature = "implication"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "implication")))]
 #[cfg(feature = "implication")]
 impl<F, T, Type> Implies<Refinement<Type, T>> for Refinement<Type, F>
 where

--- a/src/refinement/named.rs
+++ b/src/refinement/named.rs
@@ -82,7 +82,7 @@ impl<N: TypeString, T, P: StatefulPredicate<T>, R: StatefulRefinementOps<T, P>>
 }
 
 #[cfg(feature = "serde")]
-#[doc(cfg(feature = "serde"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
 mod named_serde {
     use super::*;
     use serde::{de::DeserializeOwned, Deserialize, Serialize};

--- a/src/string.rs
+++ b/src/string.rs
@@ -88,7 +88,7 @@ impl<T: AsRef<str>> Predicate<T> for Trimmed {
 }
 
 #[cfg(feature = "regex")]
-#[doc(cfg(feature = "regex"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "regex")))]
 mod regex_pred {
     use super::*;
     use crate::StatefulPredicate;

